src/client_lib/pegasus_client_impl.cpp

```diff
@@ -1302,5 +1303,61 @@ const char *pegasus_client_impl::get_error_string(int error_code) const
 {
     return (rocskdb_error == 0) ? 0 : ROCSKDB_ERROR_START - rocskdb_error;
 }
+
+bool pegasus_client_impl::detect_hotkey(const std::string &hotkey_type,
+                                        const std::string &hotkey_action,
+                                        uint64_t partition_hash,
+                                        std::string &err_info)
+{
+    if (partition_hash < 0) {
+        err_info = "error partition_num";
+        return false;
+    }
+
+    dsn::apps::hotkey_detect_request req;
+    std::string hotkey_type_check = hotkey_type;
+    std::transform(
+        hotkey_type_check.begin(), hotkey_type_check.end(), hotkey_type_check.begin(), ::tolower);
+    if (hotkey_type_check == "read") {
+        req.type = dsn::apps::hotkey_type::type::READ;
+    } else if (hotkey_type_check == "write") {
+        req.type = dsn::apps::hotkey_type::type::WRITE;
+    } else {
+        err_info = "wrong hotkey type";
+        return false;
+    }
+
+    std::string hotkey_action_check = hotkey_action;
+    std::transform(hotkey_action_check.begin(),
+                   hotkey_action_check.end(),
+                   hotkey_action_check.begin(),
+                   ::tolower);
+    if (hotkey_action_check == "start") {
+        req.action = dsn::apps::hotkey_detect_action::START;
+    } else if (hotkey_action_check == "stop") {
+        req.action = dsn::apps::hotkey_detect_action::STOP;
+    } else {
+        err_info = "wrong action type";
+        return false;
+    }
+
```

Comment:
you‘d better refactor param `check` and `transform` to one method,  such as `generate_hotkey_request`. And I think it should be the `command` logic but not put into pegasus_client_iml.cpp